library/ajax/user_settings.php

```diff
@@ -30,5 +30,5 @@
 
 // even more generic
 if ((isset($_POST['target'])) && (isset($_POST['setting']))) {
-    setUserSetting($_POST['target'], $_POST['setting']);
+    setUserSetting(attr($_POST['target']), attr($_POST['setting']));
```

Comment:
revert this. only should html escape stuff that goes out to the browser, or else we'll just twist our brains into a pretzel :)